iceoryx_hoofs/test/moduletests/test_posix_semaphore_common.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_MODULETESTS_TEST_POSIX_SEMAPHORE_COMMON_HPP
+#define IOX_HOOFS_MODULETESTS_TEST_POSIX_SEMAPHORE_COMMON_HPP
+
+#include "iceoryx_hoofs/platform/semaphore.hpp"
+#include "test.hpp"
+
+template <typename SemaphoreType>
+inline bool setSemaphoreToZeroAndVerifyValue(SemaphoreType& semaphore, const uint32_t expectedValue)
+{
+    uint32_t count = 0U;
+    for (; count + 1 < IOX_SEM_VALUE_MAX; ++count)
+    {
+        auto result = semaphore.tryWait();
+        if (result.has_error())
+        {
+            return false;
+        }
+        if (*result == false)
+        {
+            break;
+        }
+    }
+    return (count == expectedValue);
+}
+
+template <typename SemaphoreType>
+inline bool setSemaphoreValueTo(SemaphoreType& semaphore, const uint32_t value)
+{
+    for (uint32_t i = 0U; i < value; ++i)
```

Comment:
I tend to disagree here. When `value` is `uint32_t` max then `i` can be validly incremented until this value since `uint32_t` is still a valid number.
Of course the semaphore would then overflow but this would be handled inside of `post` and the user would get a `false` as return value.

Reply:
You assume that `post` is working as expected ;). But I'm also okay with leaving this as is. The worst thing that could happen is that the watchdog will be triggered but we won't have a falsely passed test.